components/locid_transform/src/directionality.rs

```diff
@@ -0,0 +1,153 @@
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    /// <div class="stab unstable">
+    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
+    /// </div>
+    pub fn try_new_unstable<P>(provider: &P) -> Result<LocaleDirectionality, LocaleTransformError>
+    where
+        P: DataProvider<DirectionalityV1Marker>
+            + DataProvider<LikelySubtagsForLanguageV1Marker>
+            + DataProvider<LikelySubtagsForScriptRegionV1Marker>
+            + ?Sized,
+    {
+        let expander = LocaleExpander::try_new_unstable(provider)?;
+        Self::try_new_with_expander_unstable(provider, expander)
+    }
+
+    icu_provider::gen_any_buffer_constructors!(
+        locale: skip,
+        options: skip,
+        error: LocaleTransformError
+    );
+
+    // TODO: Do we need to allow user-provided LocaleExpanders? Can a LocaleExpander ever not
+    // have the right amount of data? Make this public if so
+    fn try_new_with_expander_unstable<P>(
+        provider: &P,
+        expander: LocaleExpander,
+    ) -> Result<LocaleDirectionality, LocaleTransformError>
+    where
+        P: DataProvider<DirectionalityV1Marker> + ?Sized,
+    {
+        let rtl: DataPayload<DirectionalityV1Marker> =
+            provider.load(Default::default())?.take_payload()?;
+
+        Ok(LocaleDirectionality { rtl, expander })
+    }
+
+    /// Returns the script direction of the given locale.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_locid::Locale;
+    /// use icu_locid_transform::LocaleDirectionality;
+    ///
+    /// let ld = LocaleDirectionality::try_new_unstable(&icu_testdata::unstable())
+    ///     .expect("create failed");
+    ///
+    /// let locale: Locale = "en-US".parse().unwrap();
+    /// assert_eq!(ld.get(locale), icu_locid_transform::Direction::LeftToRight);
+    ///
+    /// let locale: Locale = "ar".parse().unwrap();
+    /// assert_eq!(ld.get(locale), icu_locid_transform::Direction::RightToLeft);
+    ///
+    /// let locale: Locale = "fr-Brai-FR".parse().unwrap();
+    /// assert_eq!(ld.get(locale), icu_locid_transform::Direction::Unknown);
+    /// ```
+    pub fn get(&self, locale: impl Into<Locale>) -> Direction {
+        let mut locale = locale.into();
+
+        let script = locale
+            .id
+            .script
+            .or_else(|| {
+                self.expander.maximize(&mut locale);
+                locale.id.script
+            })
+            .unwrap_or(script!("Zzzz"));
+
+        let direction = self
+            .rtl
+            .get()
+            .rtl
+            .get(&script.into_tinystr().to_unvalidated());
+        match direction {
+            Some(&direction) => <Direction as zerovec::ule::AsULE>::from_unaligned(direction),
+            // TODO: Do we make `LocaleDirectionality::get` fallible despite the DataPayload
+            // being defined to contain all scripts?
+            None => Direction::Unknown,
```

Comment:
Question: Can we assume the script of a locale is always a valid script if it exists? Then perhaps an `unreachable!` would be more fitting